onnxruntime/core/providers/webnn/builders/model_builder.h

```diff
@@ -38,11 +39,10 @@ class ModelBuilder {
   const emscripten::val& GetOpSupportLimits() const { return wnn_limits_; }
 
   void AddOperand(const std::string& name, const emscripten::val& operand);
-  const emscripten::val& GetZeroConstant(
-      const int32_t& data_type, const std::vector<uint32_t>& shape = {});
 
   template <typename T>
-  const emscripten::val& CreateOrGetScalarConstant(const int32_t& data_type, T value);
+  const emscripten::val& CreateOrGetConstant(const int32_t& data_type, T value,
+                                                   const std::vector<uint32_t>& shape = {});
```

Comment:
Can this just be a `span`? Otherwise a `vector` will need to be created on demand for the optional parameter when omitted. Granted, it's an empty vector, and so in theory no heap allocation should occur. 🤞

Reply:
Unfortunately, `emscripten::val::array(shape)` used in the function doesn't support `span`.

Reply:
The linter error is really confusing (red herring 403 error), but it appears to complain about the indentation:

```
  const emscripten::val& CreateOrGetConstant(const int32_t& data_type, T value,
                                             const std::vector<uint32_t>& shape = {});
```